python/ray/data/_internal/execution/interfaces.py

```diff
@@ -216,6 +217,9 @@ class TaskContext:
     # operator.
     task_idx: int
 
+    # The iterator of sub progress bar that task is responsible to update.
+    sub_progress_bar_iter: Optional[Iterator[ProgressBar]] = None
```

Comment:
This ideally should be treated as part of `JobContext`, because it's only living in driver side, but not task side. Added here for now for initial implementation, because we already add the hook for `TaskContext` to each stage's bulk function.

Reply:
Add a comment that this is only used on the driver side.

Reply:
This indeed looks a bit weird in TaskContext. It's more like an operator context or state. The plumbing work looks involved only all2all operators. Should we just make it that way?